### PR TITLE
threads: split CThreadImplPosix into CThreadImplLinux

### PR DESCRIPTION
**Pull request by lrusak**
This continues the cleanup and reorganization of the platform threading implementation.

Linux has some specific platform threading because we want to set the [niceness](https://man7.org/linux/man-pages/man2/nice.2.html).

I also do some simple cleanup on the implementation after the reorganization.

One noticeable change is making the usermaxpriority check static so that it only occurs once. I don't think it should be expected to recheck the rlimits when running.

### CODE REVIEW

**Comment by lrusak** [comment]
jenkins build this please

File: xbmc/platform/posix/threads/ThreadImplPosix.cpp [hunk 1]
```diff
@@ -8,172 +8,27 @@
...
-    if (setpriority(PRIO_PROCESS, m_threadID, newNice) == 0)
-      bReturn = true;
-    else
-      CLog::Log(LOGERROR, "{}: error {}", __FUNCTION__, strerror(errno));
-  }
-#endif
-
-  return bReturn;
+  CLog::Log(LOGDEBUG, "[threads] setting priority is not supported on this platform");
 }
```
Comment (Rechi) [inline xbmc/platform/posix/threads/ThreadImplPosix.cpp:34]: `xbmc/platform/posix/threads/ThreadImplPosix.cpp:34:1: warning: control reaches end of non-void function [-Wreturn-type]`

Comment (lrusak) [reply xbmc/platform/posix/threads/ThreadImplPosix.cpp:35]: Thanks!

**Comment by lrusak** [comment]
jenkins build this please

**Review by garbear — APPROVED** [review]
Nice changes.

File: xbmc/platform/linux/threads/ThreadImplLinux.cpp [hunk 2]
```diff
@@ -101,7 +87,7 @@ std::unique_ptr<IThreadImpl> IThreadImpl::CreateThreadImpl(std::thread::native_h
 }
 
 CThreadImplLinux::CThreadImplLinux(std::thread::native_handle_type handle)
-  : IThreadImpl(handle), m_threadID(GetCurrentThreadPid_())
+  : IThreadImpl(handle), m_threadID(gettid())
```
Comment (bkuhls) [inline xbmc/platform/linux/threads/ThreadImplLinux.cpp:103]: sent https://github.com/xbmc/xbmc/pull/21215 to fix a build error with uClibc

**Pull request merged**